### DISCUSSION TITLE
I convert a 3M image to webp，it found error : partition 0 overflow (> 512K)

### DISCUSSION THREAD

**Discussion by tpwonline** [ImageMagick/ImageMagick#4594] [Help]
raw image : a 3M jpg , 7360 x 4912 , 72dpi
my version : ImageMagick 6.9.10-68 Q16 x86_64 2020-04-01
how can i resolve this problem.

**Comment by dlemstra** [comment]
Can you attach the image to this question? And can you reproduce this with the most recent version of ImageMagick?

**Comment by taylorcjensen** [comment]
I'm encountering the same issue, below are the args I'm using. ImageMagick v6. [Here's](https://images.home.tcj.one/5d3b7ae83467baebc7620f78829631e313c89fe324400d69c851b70a734bc9d6.jpg) a sample file.

```
argumentList: [
            Variables.file.FullName,
            '-resize',
            '16383x16383>',
            '-quality',
            '85',
            '-define',
            'webp:use-sharp-yuv=true',
            '-define',
            'webp:auto-filter=true',
            '-define',
            'webp:method=6',
            '-define',
            'webp:preprocessing=0',
            '-define',
            'webp:partitions=3',
            '-define',
            'webp:thread-level=1',
            // '-define',
            // 'webp:lossless=true',
            output,
        ]
```
These are my policy limits:
```
<policymap>
  <policy domain="resource" name="memory" value="2000MiB"/>
  <policy domain="resource" name="map" value="2000MiB"/>
  <policy domain="resource" name="width" value="100KP"/>
  <policy domain="resource" name="height" value="100KP"/>
  <policy domain="resource" name="area" value="1000MP"/>
  <policy domain="resource" name="disk" value="100GiB"/>
</policymap>
```

**Comment by snibgo** [comment]
The link to the sample file is bad:
```
C:\web\im>wget https://images.home.tcj.one/5d3b7ae83467baebc7620f78829631e313c89fe324400d69c851b70a734bc9d6.jpg

--2023-11-17 13:22:48--  https://images.home.tcj.one/5d3b7ae83467baebc7620f78829631e313c89fe324400d69c851b70a734bc9d6.jpg
Resolving images.home.tcj.one (images.home.tcj.one)... 71.125.40.14
Connecting to images.home.tcj.one (images.home.tcj.one)|71.125.40.14|:443... connected.
GnuTLS: The request is invalid.
Unable to establish SSL connection.
```

**Comment by taylorcjensen** [comment]
My mistake, I'd made a typo on my image host. The previous link should work fine now.

**Comment by urban-warrior** [comment]
The exception is thrown by the WEBP delegate library and can be reproduced by the `cwebp` utility.  First reduce the image size with ImageMagick to something less than 16383 pixels.  Next, convert with `cwebp`:

```
Saving file 'image.webp'
Error! Cannot encode picture as WebP
Error code: 6 (PARTITION0_OVERFLOW: Partition #0 is too big to fit 512k.
To reduce the size of this partition, try using less segments with the -segments option, and eventually reduce the number of header bits using -partition_limit. More details are available in the manual (`man cwebp`))
```

To resolve this issue, post to the WEBP developers and let us know what they advise.

**Comment by taylorcjensen** [comment]
I'm already attempting to resize in that command though, can I not do those steps at the same time? When I use the same command to resize to something smaller like 12K x 12K it works fine.

**Comment by urban-warrior** [comment]
The issue lies within the WEBP delegate library rather than ImageMagick. Your options are to either decrease the image dimensions to a maximum of 12,000 x 12,000 pixels or report the issue to the WEBP developers for a potential patch. If the patch is implemented, you can then compile ImageMagick with the updated release of the WEBP delegate library, enabling ImageMagick to accurately generate WEBP images with dimensions up to 16,383 x 16,383 pixels.

**Comment by audioscavenger** [comment]
hi, still an issue in 2025. Just so you know. Why is there a 12,000 pixels limit anyway??

**Reply by fmw42** [reply]
You have to ask the webp developers.

**Reply by audioscavenger** [reply]
thanks! but this is a lost cause, how do you even contact a Google dev? I use AVIF instead, that's the best solution.

**Reply by jzern** [reply]
> hi, still an issue in 2025. Just so you know. Why is there a 12,000 pixels limit anyway??

The resolution limit for WebP is 16383x16383, but there are some limitations in the underlying VP8 (lossy) bitstream that can cause encode failures depending on the content. See [`-partition_limit`](https://developers.google.com/speed/webp/docs/cwebp) in the cwebp documentation. Other options (like `-m`, the 'method' or effort the encoder uses) will also affect the encoder's ability to come in under this limit.

The issue has to do with the size of the compressed partition, which contains the DCT coefficients for each block. The first partition in VP8 is limited to 512K bytes (19 bits). So with high quality and smaller blocks we can push up into this limit pretty quickly with high resolution images.

> [...] how do you even contact a Google dev?

Feel free to post any questions about webp to our [mailing list](https://chromium.googlesource.com/webm/libwebp#discuss), webp-discuss@webmproject.org.
